src/rolling/directory.rs

```diff
@@ -108,6 +113,32 @@ impl Directory {
         file.seek(SeekFrom::Start(0u64)).await?;
         Ok(file)
     }
+
+    pub async fn sync_data(&self, files: std::ops::Range<u64>) -> io::Result<()> {
+        let file_sync_tasks =
+            futures_util::stream::iter(files.into_iter()).map(|file_number| async move {
+                let filepath = filepath_for_number(&self.dir, file_number);
+                match OpenOptions::new()
+                    .read(true)
+                    .create(false)
+                    .open(&filepath)
+                    .await
+                {
+                    Ok(file) => file.sync_data().await,
+                    Err(err) if err.kind() == io::ErrorKind::NotFound => Ok(()),
+                    e => e.map(|_| ()),
+                }
+            });
+
+        // We do this to not have more than a few fd open at once.
+        // Not doing that breaks tests, but is unlikely to cause issues in practice
+        let mut futures = file_sync_tasks.buffer_unordered(16);
```

Comment:
I don't think it is worth doing this concurrently at all. In which configuration would imagine having to fsync a lot of files? How many is a lot, and what parameters would see that happen?

Reply:
in quickwit use case where we regularly flush, never? It should happen only if you only call flush once in a while (or in tests where files are 128KiB, but we don't really care)